### ISSUE TITLE
Option to send or to make a new line when pressing the enter key in the virtual/hardware keyboard

### ISSUE THREAD

**Issue by TheMythPT** [RocketChat/Rocket.Chat.ReactNative#746]
The app could have an option to `send on enter` or `create new line` when using the enter key.
By default, IMHO, it should respect the settings we have in our profile (I've set in to send via web interface).

**Comment by diegolmello** [comment]
@TheMythPT Nice!

**Comment by pranavpandey1998official** [comment]
@TheMythPT if he chooses send on enter how will he make a new line if he wants (on Android especially), I tried but couldn't do it

**Comment by TheMythPT** [comment]
The idea is the following:
1. The app should use the default set on browser and, iirc, the default is to send on Enter.
2. The option on the app would override the web settings, so basically the behaviour inside the app would differ from the browser behaviour.

**Comment by diegolmello** [comment]
@TheMythPT How would the user add a new line?
Do you have an example app that does it? Twitter, Facebook, etc?

**Comment by TheMythPT** [comment]
Like I've said above
1. If the `Send on Enter` is enabled, there's no option to add a newline.
2. If `Send on Enter` is disabled, current behaviour is used.

**Comment by diegolmello** [comment]
@TheMythPT I've tried Slack, WhatsApp, Twitter and Discord.
None of them have this behaviour.

**Comment by TheMythPT** [comment]
Makes sense, although those apps are really a bit crappy without almost any possible customisation.

Check [here](https://nextcloud.cloudbnc.tk/s/wKybmY6HtFJ6SRg)

**Comment by diegolmello** [comment]
@TheMythPT An example! Now we're talking :)

**Comment by TheMythPT** [comment]
> @TheMythPT An example! Now we're talking :)

Usually I don't outsource my sources. I like people to think that was my idea 😋

**Comment by TheMythPT** [comment]
@diegolmello also, your [WhatsApp](https://nextcloud.cloudbnc.tk/s/3wddYXocz48iAWy) must be very outdated! :stuck_out_tongue_closed_eyes:

**Comment by diegolmello** [comment]
@TheMythPT My WhatsApp is updated on iOS and I don't have this feature, but it's nice to know it exists!

**Comment by TheMythPT** [comment]
@diegolmello, you should already know that good stuff is just available to Android users!
😝

**Comment by andrewgribben** [comment]
Just a quick +1 for this feature, I use an iPad with magic keyboard a lot and it's frustrating that there is no keyboard only way of sending a message. Return, Shift+return and alt+return all do the same thing so it'd be nice of one those could be repurposed

**Comment by maiksd** [comment]
Agreed, and I'd like to add that I can't even tap the send button without moving the iPadOS keyboard controls out of the way first, for every single message. There really should be a way to send via keyboard, like Option-Return or Cmd-Return.

![IMG_5135](https://user-images.githubusercontent.com/903073/177537276-b15aac35-620c-4b89-be0a-f2676feb279f.jpeg)

**Comment by diegolmello** [comment]
@maiksd How can I reproduce that?
I was no able to open those keyboard controls on top of send button.

**Comment by diegolmello** [comment]
That's an OS thing and you can move it around.

https://user-images.githubusercontent.com/804994/177596867-462310a9-34cc-4767-8dd0-c893893c7493.mp4

**Comment by maiksd** [comment]
Sure you can, and you must to send a message. But it will reappear in the lower right corner again and again anyway, so that's not a solution.

**Comment by diegolmello** [comment]
> But it will reappear in the lower right corner again and again anyway, so that's not a solution.

No, it's going to appear wherever you placed for the last time.

Anyway, you can just turn that off by disabling shortcuts and predictive.
You don't want to show it.

**Comment by maiksd** [comment]
Sure, I can deal with it by moving it around or using iOS settings that I don't want to use. Disabling shortcuts loses me the ability to invoke dictation, for example.

Why not improve the application and do it like e.g. Slack does it (return defaults to send, option-return for new line)?

**Comment by diegolmello** [comment]
I was just trying to help you use the app.
This issue is open yet because we're looking for contributors and this is a wanted feature.
Are you willing to contribute?

**Comment by deepak0x** [comment]
have checked that the feature is not implemented currently...
can I work on this then...
